src/cast_core/src/RunCommand.cpp

```diff
@@ -232,11 +354,17 @@ struct actor::RunCommand::PhaseConfig {
 
     bool throwOnFailure;
     std::vector<std::unique_ptr<DatabaseOperation>> operations;
+    InstanceTypeFilter onlyRunInInstancesFilter;
 };
 
 void actor::RunCommand::run() {
     for (auto&& config : _loop) {
         for (auto&& _ : config) {
+            // Run ops when there is no OnlyRunInInstances option or when one of the options matches
+            if(!config->onlyRunInInstancesFilter.shouldRun()) {
```

Comment:
Also, thanks for adding such great test coverage!

Reply:
@aggrand Hey, I also thought about that and tried it. Was getting failures in NOP phases, because the config cannot be dereferenced in nop phases. So I decided to move it back in. I just discovered that there's a `isNop` check, so I moved it out again! 

Quick test: https://spruce.mongodb.com/version/62696fed32f41723a8313d38/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC